antarest/study/storage/abstract_storage_service.py

```
@@ -45,12 +50,77 @@
 from antarest.study.storage.rawstudy.model.filesystem.config.model import Simulation
 from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy, StudyFactory
+from antarest.study.storage.rawstudy.model.filesystem.folder_node import FolderNode
+from antarest.study.storage.rawstudy.model.filesystem.inode import INode
+from antarest.study.storage.rawstudy.model.filesystem.lazy_node import LazyNode
+from antarest.study.storage.rawstudy.model.filesystem.matrix.input_series_matrix import InputSeriesMatrix
 from antarest.study.storage.rawstudy.model.helpers import FileStudyHelpers
 from antarest.study.storage.utils import extract_output_name, fix_study_root, remove_from_cache
 
 logger = logging.getLogger(__name__)
 
 
+def _assert_not_folder_node(file_node: INode[t.Any, t.Any, t.Any]) -> None:
+    if isinstance(file_node, FolderNode):
+        raise NotFilePath("Node is a folder node.")
+
+
+def _parse_file_from_link(file_node: INode[t.Any, t.Any, t.Any], link: str) -> bytes:
+    if not isinstance(file_node, InputSeriesMatrix):
+        raise ShouldNotHappenException(f"Node {file_node.config.path} has a link but is not an InputSeriesMatrix.")
+    tmp_dir = tempfile.TemporaryDirectory()
+    # target path to save the file
+    target_path = Path(tmp_dir.name) / file_node.config.path.name
+    # use `.tsv` as suffix for the file
+    target_path = target_path.with_suffix(".tsv")
+    # retrieve the file content
+    file_node.parse_from_link(link).dropna(how="any", axis=1).to_csv(
+        target_path,
+        sep="\t",
+        index=True,
+        header=True,
+        float_format="%.6f",
+    )
+    output = target_path.read_bytes()
+    tmp_dir.cleanup()
+    return output
+
+
+def _extract_original_file_from_unarchived_study(
+    file_node: INode[t.Any, t.Any, t.Any], metadata: T, url: str
+) -> OriginalFile:
+    _assert_not_folder_node(file_node)
+    if isinstance(file_node, LazyNode) and file_node.get_link_path().is_file():
```

The service layer must not know about the concept of "link", this must be handled inside the node implementation, by a way or another.
(but I think it's more or less a duplicate of the comment of Martin above about using the `parse(return_dataframe=True)` method)

---

We need to isolate this specific case or most code would be moved into `inode.py`

---

Then we need to have an abstract method in `INode` and an implementation in `LazyNode`.
The `if isinstance(file_node, LazyNode)` also is a clear hint that inheritance is relevant here. For example, we can have
```python
class INode:
    def get_content() -> bytes:
        return self.config.path.read_bytes()
```
```python
class LazyNode:
    def get_content() -> bytes:
        # do stuff to handle links
```

---

Also, `FolderNode` implementation should raise an exception.